tests/functional/test_revert_context_manager.py

```diff
@@ -43,32 +63,59 @@ def test_revert_msg_raises_partial(owner, reverts_contract_instance):
 @geth_process_test
 def test_dev_revert(owner, reverts_contract_instance, geth_provider):
     """
-    Test catching transaction reverts and asserting on a dev message written in the contract source
-    code.
+    Test matching a contract dev revert message with a supplied dev message.
     """
-
     with reverts(dev_message="dev: error"):
         reverts_contract_instance.revertStrings(2, sender=owner)
 
 
+@geth_process_test
+def test_dev_revert_pattern(owner, reverts_contract_instance, geth_provider):
+    """
+    Test matching a contract dev revert message with a supplied dev message pattern.
+    """
+    with reverts(dev_message=re.compile(r"dev: err\w+")):
+        reverts_contract_instance.revertStrings(2, sender=owner)
+
+
 @geth_process_test
 def test_dev_revert_fails(owner, reverts_contract_instance, geth_provider):
     """
-    Test that ``AssertionError`` is raised if the supplied dev message does not match the actual
-    dev message found in the contract at the source of the revert.
+    Test that ``AssertionError`` is raised if the supplied dev message and the contract dev message
+    do not match.
     """
+    with pytest.raises(AssertionError):
+        with reverts(dev_message="dev: foo"):
+            reverts_contract_instance.revertStrings(2, sender=owner)
 
+
+@geth_process_test
+def test_dev_revert_partial_fails(owner, reverts_contract_instance, geth_provider):
+    """
+    Test that ``AssertionError`` is raised if the supplied dev message and the contract dev message
+    do not match exactly.
+    """
     with pytest.raises(AssertionError):
         with reverts(dev_message="dev: foo"):
             reverts_contract_instance.revertStrings(2, sender=owner)
 
 
 @geth_process_test
-def test_both(owner, reverts_contract_instance, geth_provider):
+def test_dev_revert_pattern_fails(owner, reverts_contract_instance, geth_provider):
     """
-    Test catching transaction reverts and asserting on the revert reason as well as a dev message
-    written in the contract source code.
+    Test that ``AssertionError`` is raised if the contract dev message does not match the supplied
+    dev revert pattern.
     """
+    with pytest.raises(AssertionError):
+        with reverts(dev_message=re.compile(r"dev: [^ero]+")):
+            reverts_contract_instance.revertStrings(2, sender=owner)
+
 
+@geth_process_test
+def test_both(owner, reverts_contract_instance, geth_provider):
```

Comment:
```suggestion
def test_both_message_and_dev_str(owner, reverts_contract_instance, geth_provider):
```

or something a ta more